spyder/plugins/updatemanager/tests/test_update_manager.py

```diff
@@ -51,9 +51,10 @@ def test_updates_appenv(qtbot, mocker, version):
         return_value=("conda-forge", "https://conda.anaconda.org/conda-forge")
     )
 
-    um = UpdateManagerWidget(None)
-    um.start_check_update()
-    qtbot.waitUntil(um.update_thread.isFinished)
+    with caplog.at_level(logging.DEBUG, logger='spyder.plugins.updatemanager'):
```

Comment:
What does this exactly do? Please add a comment on top of this line to explain it.

Reply:
The `caplog` fixture (`pytest` builtin fixture) tells `pytest` to capture output from `logging`. The context, of course, limits this to the enclosed context, i.e. just the following `um.start_check_update`. Additionally, for the context, the specified logger's logging level changed to the specified level. Note, however, that while the logging output is captured by `pytest`, it will not be _reported_ by `pytest` unless this particular test fails; and the report will only be displayed at the end of the test run in a "captured logging" section.